src/main/java/org/fungover/mmotodo/team/TeamService.java

```diff
@@ -0,0 +1,124 @@
+package org.fungover.mmotodo.team;
+
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import jakarta.validation.Valid;
+import org.fungover.mmotodo.exception.TeamNotFoundException;
+import org.fungover.mmotodo.task.Task;
+import org.fungover.mmotodo.task.TaskRepository;
+import org.fungover.mmotodo.user.User;
+import org.fungover.mmotodo.user.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+
+@Service
+public class TeamService {
+    private final TeamRepository teamRepository;
+
+    private final UserRepository userRepository;
+
+    private final TaskRepository taskRepository;
+    @Autowired
+    public TeamService(TeamRepository teamRepository, UserRepository userRepository, TaskRepository taskRepository) {
+
+        this.teamRepository = teamRepository;
+        this.userRepository = userRepository;
+        this.taskRepository=taskRepository;
+    }
+
+    //get all teams
+    public List<Team> getAllTeams(){
+        return teamRepository.findAll();
+    }
+ public Team getTeamById(int teamId){
+        return teamRepository.findById(teamId).orElseThrow(() -> new TeamNotFoundException("Team not found"));
+ }
+
+    @Transactional
+    public String deleteTeam(int teamId){
+        Team team = teamRepository.findById(teamId).orElseThrow(() -> new TeamNotFoundException("Team not found"));
+       teamRepository.delete(team);
+     return "Team with id:" + teamId + " deleted";
+ }
+
+
+    @Transactional
+    public Team createTeam( @Valid TeamDto teamCreate) {
+        // Check if the team with the same name already exists
+        if (teamRepository.existsByName(teamCreate.name())) {
+            throw new RuntimeException("Team already exists");
+        }
+
+            Team team = new Team();
+            team.setName(teamCreate.name());
+            team.setUsers(new ArrayList<>());
+            team.setTasks(new ArrayList<>());
```

Comment:
Why do you set an empty list of users and tasks here?

Reply:
Start with Ctrl-Alt-L and also check that the reformat of code is run automatically before doing a commit.